tutorials/04 - DistortionEffectDemo.md

```diff
@@ -0,0 +1,138 @@
+# 04 - Distortion Effect
+
+In this tutorial, we're going to analyze the structure of an audio effect application, by applying distortion to a looping audio file.
+
+## Main Tracktion Engine Classes
+#### Engine
+Every tracktion Engine app starts with an 'Engine' instance. This performs the initialisation and shutdown of the engine.
+
+#### Edit
+The 'Edit' is the container for a playable arrangement. it holds tracks, tempo sequences, Racks etc.
+
+#### Track
+'Edit' contains a list of 'Track's. This project makes use of the 'AudioTrack' which outputs audio.
+
+#### Clip
+'Clip' is the base class for clips that live in 'Track's. 'Clip's can be MIDI, audio, step or the edit-in-an-edit 'editClip.'
+
+
+## Main JUCE Classes
+#### Plugin
+
+## DistortionEffectDemo.h
+
+
+## DistortionPlugin Class
+
+### DistortionPlugin() Constructor
+
+- In the constructor, we initialise the undo manager to enable users to undo plugin value changes, and refer each plugin value to the undo manager.
+```
+auto um = getUndoManager();
+gainValue.referTo (state, IDs::gain, um, 1.0f);
+```
+
+- We then initialise each plugin parameter and attach them to the values they represent.
```

Comment:
Probably better to more accurate and rather than "to the values they represent" say something like "We create an AutomatableParameter, assign it to our gainParam member. `addParam` registers the parameter with the `Plugin`".
Then something like the `attachToCurrentValue` call attaches the `gainValue` `CachedValue` member to the gain parameter.